dashboard/views.py

```diff
@@ -150,3 +152,37 @@ def post(self, request):
         return Response(
             data=enrollment.json
         )
+
+
+class UnEnrollPrograms(APIView):
+    """
+    api that unenroll user from one ore more programs
+    """
+    authentication_classes = (
+        authentication.SessionAuthentication,
+        authentication.TokenAuthentication,
+    )
+    permission_classes = (permissions.IsAuthenticated, )
+
+    def post(self, request):
+        """
+        unenroll from MM program(s)
+        """
+        program_ids_serializer = UnEnrollProgramsSerializer(data=request.data)
+        program_ids_serializer.is_valid(raise_exception=True)
+        program_ids = program_ids_serializer.data['program_ids']
+        program_enrollments = ProgramEnrollment.objects.filter(
+            program_id__in=program_ids,
+            user=request.user
+        )
+        response = []
+        for program_enrollment in program_enrollments:
+            response.append({
+                'program_id': program_enrollment.program_id,
+                'title': program_enrollment.program.title
+            })
+            program_enrollment.delete()
```

Comment:
Are we confident in the side effects of outright deleting the `ProgramEnrollment` here? We generally don't delete data so we should verify that no parts of our application (particularly any scheduled tasks). Easy ones to find will be any `ProgramEnrollment.objects.get()` queries that aren't error-handled, but I suspect there may be more difficult edge cases to tease out. We may want to consider just marking these and inactive and filtering them out instead, although I realize that might be just as laborious.

Reply:
It's been my practice to delete ProgramEnrollment records when users wish to have their account deleted. I know this isn't exactly the same though, so I guess there are more cases to consider. 

If we take your suggestion of marking them inactive, will we need a migration, along with updating the code to filter out inactive ProgramEnrollments?

Reply:
Yes, well need both. I guess my point was, the amount of effort to do that is probably equal to or possibly greater than the amount of effort needed to shore up existing code. that doesn't handle situations where the record doesn't exist and is expected to, because those are probably the minority of all `ProgramEnrollment` code.

TLDR: deleting is **probably** still the best option (dragons aside), but we definitely need to test and/or audit at least the obvious code points that use these records to head off any errors. We probably won't catch them all, but unhandled `get(user=?, program=?)` calls are obvious candidates for where you'd hit that kind of error.

Reply:
i tested all pages after removing all or single program, they are working fine.

Reply:
@rhysyngsun any thoughts

Reply:
@amir-qayyum-khan you should test code that isn't on pages too, such as celery tasks, that query this.